src/graph/heterograph.h

```diff
@@ -251,6 +273,8 @@ class HeteroGraph : public BaseHeteroGraph {
   FlattenedHeteroGraphPtr FlattenImpl(const std::vector<dgl_type_t>& etypes) const;
 };
 
+const int64_t SHARED_MEM_METAINFO_SIZE_MAX = 1024 * 16;  // at most 16k
```

Comment:
Is this enough with list of ntypes and etypes string? Or use MemoryStringStream and create the shared memory with the size of the backup string? (Sorry for 反复横跳)

Reply:
I have some questions
>     Raise error when calling shared_memory if graph index is on gpu
it returns a copy of the graph index in shared memory, no matter where the original graph is.

>     Formats should not be specified, just saving all the materialized formats
then users have to manually materialize formats? one of the purposes of this pr is to avoid such functions like `prepare_mp` in multi gpu version of graphsage.

Reply:
> Is this enough with list of ntypes and etypes string? Or use MemoryStringStream and create the shared memory with the size of the backup string? (Sorry for 反复横跳)

Now the number of etypes is limited by the number of file descriptors, which means a graph in shared memory can have at most 1k~1.6k etypes

Reply:
I think it's fine to add TODOs for now. And let's discuss later when the cuda_graph pr merged.